pkg/controllers/migration/crd/controller.go

```diff
@@ -62,10 +62,20 @@ func (c *Controller) Reconcile(ctx context.Context, crd *apiextensionsv1.CustomR
 	}); !ok && nodeClassGvk.Kind != crd.Spec.Names.Kind {
 		return reconcile.Result{}, nil
 	}
+
 	// If v1beta1 is not stored, no-op
 	if !lo.Contains(crd.Status.StoredVersions, "v1beta1") {
 		return reconcile.Result{}, nil
 	}
+	// We only want to reconcile against CRDs that have v1 as the storage version.
+	// Note: we don't need to check this in the resource migration controller since the conversion webhook drops the
+	// annotation when applying the v1 resource with the annotation.
+	if storageVersion, found := lo.Find(crd.Spec.Versions, func(v apiextensionsv1.CustomResourceDefinitionVersion) bool {
+		return v.Storage
+	}); !found || storageVersion.Name != "v1" {
```

Comment:
Should we add some sort of log here to let the user know that they haven't upgraded their CRD and that we expect it to be?

Reply:
Good call, I'll add one 👍